src/Model/CoffeeShop.kt

```diff
@@ -0,0 +1,14 @@
+
+internal data class CoffeeShop(val coordinates: Coordinates, val name: String) {
+    var distanceFromUser: Double? = null
```

Comment:
- It can lead to unintended side-effects;
- It's not thread safe. As the object can be changed, multiple threads might try to update it at the same time, or they might see separate versions of it;
- It is easier to think about immutable objects since you don't have to think about the ways they might change over time (this can help in debugging for example).